funnel/models/rsvp.py

```diff
@@ -40,23 +38,60 @@ class Rsvp(TimestampMixin, db.Model):
         'state',
         db.CHAR(1),
         StateManager.check_constraint('state', RSVP_STATUS),
-        default=RSVP_STATUS.A,
+        default=RSVP_STATUS.MAYBE,
         nullable=False,
     )
     state = StateManager('_state', RSVP_STATUS, doc="RSVP answer")
 
+    @with_roles(call={'auth'})
+    @state.transition(
+        None,
+        state.YES,
+        title=__("Going"),
+        message=__("Your RSVP has been saved"),
+        type='primary',
+    )
+    def rsvp_yes(self):
+        pass
+
+    @with_roles(call={'auth'})
+    @state.transition(
+        None,
+        state.NO,
+        title=__("Not going"),
+        message=__("Your RSVP has been saved"),
+        type='dark',
+    )
+    def rsvp_no(self):
+        pass
+
+    @with_roles(call={'auth'})
+    @state.transition(
+        None,
+        state.MAYBE,
+        title=__("Maybe"),
+        message=__("Your RSVP has been saved"),
+        type='accent',
+    )
+    def rsvp_maybe(self):
+        pass
+
     @classmethod
-    def get_for(cls, project, user, create=False):
+    def get_for(cls, project, user, create=False, session_add=False):
         if user:
             result = cls.query.get((project.id, user.id))
             if not result and create:
                 result = cls(project=project, user=user)
-                db.session.add(result)
+                # we need to show the transition form on the project page but
+                # every person who opens the project page may not want to RSVP.
+                # so this flag lets us create an Rsvp object without adding it to the db.
+                if session_add:
+                    db.session.add(result)
```

Comment:
This is not safe. SQLAlchemy has session cascades, so if `User` or `Project` has a backref to `Rsvp`, then it will be automatically added to the session. Instead, pass `create=False` if you don't want to create a database object. Remove this code.